### DISCUSSION TITLE
Sly completion via Corfu and kind-icons doesn't show icons

### DISCUSSION THREAD

**Discussion by mdbergmann** [joaotavora/sly#631] [General]
Hello.

I switched completion from company to Corfu with kind-icons.
I get icons in Elisp files, but in Common Lisp files there are no icons in the completions.
Now I don't know where to start troubleshooting.
Any hints?

**Comment by joaotavora** [comment]
Corfu isn't picking up the completion kind hints appropriately, or there aren't any.  It seems there aren't any: `sly--completions-complete-symbol-1` in `lib/sly-completion.el` doesn't have a `:company-kind` hint.  Look there, sorry I can't be of more help.

**Reply by mdbergmann** [reply]
Strange, I see stuff like 'function', 'macro', 'var', etc. in the completion overlay. Just no icons for it.
<img width="389" alt="Bildschirmfoto 2024-03-22 um 16 33 50" src="https://github.com/joaotavora/sly/assets/2143978/594054e4-1d89-4dc8-9f2a-407bc17c69f1">

OK, thanks anyhow. I'll have a closer look.

**Reply by joaotavora** [reply]
Fairly sure that's `:annotation-function` doing that.  For company/corfu fanciness, you need `:company-kind`.  Might be wrong though.

**Reply by mdbergmann** [reply]
Yeah, we need `:company-kind`, confirmed with kind-icons maintainer.

**Reply by mdbergmann** [reply]
Is it a big effort to provide `:company-kind` would you say?

**Reply by joaotavora** [reply]
Hmmm, no I wouldn't think so.  It's basically the same as annotation but translating it to whatever :company-kind expects.  Variables, Funtions and Macros, etc Don't know if company distinguishes between functions and macros though
